core/database_api.php

```diff
@@ -309,9 +309,10 @@ function db_query_bound() {
  * @param integer $p_limit     Number of results to return.
  * @param integer $p_offset    Offset query results for paging.
  * @param boolean $p_pop_param Set to false to leave the parameters on the stack
+ * @param boolean $p_silent    Set to true to supress error messages if query fails
```

Comment:
As I said to changes in DB structure: I simply don't know how to make atomic operations in Mantis.

How you survived so far: you present UI to edit bug with secret unique token. If user posts an edit, but token mismatch last used token - you tell user that someone else has edited the bug, and then the process repeats.

I don't see how this logic could be used with automating reporting via API. 

One can use a similar technique to insert bug, then check if another bug (with same alias) already exists, then delete dup, update votes. But wouldn't it be a bit tooo complex? Instead you can just run a single INSERT and fail on uniqueness constraint broken, then update.